automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodConfigSpec.scala

```diff
@@ -132,34 +227,30 @@ class MethodConfigSpec extends FreeSpec with Matchers with WebBrowserSpec with W
 
           withWebDriver { implicit driver =>
             withSignIn(user) { _ =>
-              val variables = ListMap(
-                "w.t.inString" -> "\"test\"",
-                "w.t.inFloat"-> "1.5",
-                "w.t.inInt" -> "2",
-                "w.t.inBoolean" -> "true",
-                "w.t.inFile" -> "\"gs://foo/bar\"",
-                "w.t.inStringArray" -> """["foo","bar"]""",
-                "w.t.inStringArray2" -> """["say \"hi\"!"]""",
-                "w.t.inStringMap" -> """{"foo":"bar"}"""
-              )
-              val unmatchedVariables = ListMap(
-                "unmatched.variable.name" -> "\"surprise!\""
-              )
-
+              // Go to method config page
               val configPage = new WorkspaceMethodConfigDetailsPage(projectName, workspaceName, projectName, configName).open
 
+              // We should not be in edit mode
               configPage.isEditing shouldBe false
-              variables.keys.foreach(name => configPage.readFieldValue(name) shouldBe "")
 
-              val inputs = variables ++ unmatchedVariables map {
-                case (name, json) => (name, json)
-              }
-              configPage.populateInputsFromJson(generateInputsJson(inputs))
+              val inputs = wdlInputsBase ++ refInputsJsonFormat
+
+              // All the input fields should be empty
+              inputs.keys.foreach(name => configPage.readFieldValue(name) shouldBe "")
+
+              // Populate input fields from a json containing all the field values and some fields that don't exist
+              configPage.populateInputsFromJson(generateInputsJson(inputs ++ unmatchedVariables))
 
+              // We should have been automatically switched to edit mode
               configPage.isEditing shouldBe true
-              variables foreach {
+
+              // for each input, (not included the fields that didn't exist), we should have the correct value
+              // for referential inputs ($this._ or $workspace._), it should have removed the "$"
```

Comment:
Except it looks like you modified `generateInputsJson` to remove the leading "$" so there's nothing left for FC to remove.

Reply:
whoops